pkg/api/client.go

```diff
@@ -57,13 +55,13 @@ type httpClient struct {
 
 // URL resolves an endpoint to a fully qualified URL.
 func (c *httpClient) URL(ep string) *url.URL {
-	// IMPLICIT: c.endpoint.Path cannot end with "/" because it was stripped off in NewClient
-	// IMPORTANT: Do not use Path.Join as it will not properly preserve trailing slashes
-
-	p := c.endpoint.Path + "/" + strings.TrimLeft(ep, "/")
-	u := *c.endpoint
-	u.Path = p
-	return &u
+	u, err := c.endpoint.Parse(ep)
+	if err != nil {
+		// If code panics here, the caller needs to verify it's input before
+		// passing it on to the `Client.URL(endpoint string)` function.
+		panic(err)
```

Comment:
is panic used here instead of an error to not have to refactor all callers to handle an error? 
(always suspect of panics)

Reply:
Yes-ish. It would make the callers all really obnoxious because they usually just take `URL(...).String()` as a one-liner; but also the endpoint used to be a constant so we always knew it was a valid URL. I think that is a good pre-condition to maintain.